File: Episode08-Explosions/Love2D/shield.lua
local Shield = {} 	-- simple table for Shield. As there is only one, class not required
local Shared = require "shared"

Shield.barLength = 100
Shield.barHeight = 10
Shield.lineColour = Shared.WHITE
Shield.fillColour = Shared.DARKBLUE
Shield.pct = 100
Shield.outlineRect = { x = 5, y = 5, w = Shield.barLength, h = Shield.barHeight}
Shield.fillRect = { x = 5, y = 5, w = Shield.barLength, h = Shield.barHeight}

function Shield.update(value)
	if value < 0 then
		value = 0
	end	
	Shield.pct = value
	Shield.fillRect.w = (Shield.pct / 100) * Shield.barLength 
end

function Shield.draw()
	love.graphics.setColor(Shield.fillColour) 	-- fill background of whole rectangle
	love.graphics.rectangle("fill", Shield.fillRect.x, Shield.fillRect.y, Shield.barLength, Shield.barHeight)
	love.graphics.setColor(Shared.GREEN)				-- fill background of active shield value
	love.graphics.rectangle("fill", Shield.fillRect.x, Shield.fillRect.y, Shield.fillRect.w, Shield.fillRect.h)
	love.graphics.setColor(Shield.lineColour)	-- draw line round whole rectangle
	love.graphics.rectangle("line", Shield.outlineRect.x, Shield.outlineRect.y, Shield.outlineRect.w, Shield.outlineRect.h)
end

return Shield

File: Epidode08-Explosions/Pygame/shield.py
import pygame, shared

class Shield():
	def __init__(self, x, y, line_colour, fill_colour) -> None:
		self.BAR_LENGTH = 100
		self.BAR_HEIGHT = 10
		self.line_colour = line_colour
		self.fill_colour = fill_colour
		self.pct = 0
		self.outline_rect = pygame.Rect(x, y, self.BAR_LENGTH, self.BAR_HEIGHT)
		self.fill_rect = pygame.Rect(x, y, self.BAR_LENGTH, self.BAR_HEIGHT)	
		
	def update(self, value) -> None:
		if value < 0:
			value = 0	
		self.pct = value
		self.fill_rect.width = (self.pct / 100) * self.BAR_LENGTH
	
	def draw(self) -> None:
		pygame.draw.rect(shared.screen, self.fill_colour, self.fill_rect)
		pygame.draw.rect(shared.screen, self.line_colour, self.outline_rect, 2) #2 pixel thick line